src/main/java/de/uni_stuttgart/it_rex/course/service/written/CourseService.java

```diff
@@ -212,7 +221,38 @@ public CourseDTO patch(final CourseDTO courseDTO) {
             getSpecFromActiveAndExample(activeOnly, courseExample);
         courses = courseRepository.findAll(spec);
 
-        return courseMapper.toDTO(courses);
+        List<CourseDTO> courseDtos = courseMapper.toDTO(courses);
+        addRole(courseDtos);
+        return courseDtos;
+    }
+
+    /**
+     * Method finds all user courses and filters them by given parameters.
+     * additional filters: <p>
+     * - remove if {@link COURSEROLE#PARTICIPANT} && unpublished
+     *
+     * @param publishState Publish state of course.
+     * @param activeOnly   Set true to only include active courses (current time
+     *                     between course start and end date + offset).
+     * @return A list of courses that fit the given parameters.
+     */
+    public List<CourseDTO> findUserCourses(
+        final Optional<PUBLISHSTATE> publishState,
+        final Optional<Boolean> activeOnly) {
+        LOGGER.debug("Request to get user Courses");
+
+        List<Course> courses;
+        Example<Course> courseExample = Example.of(applyFiltersToExample(publishState));
+        Specification<Course> spec = getSpecFromActiveAndExample(activeOnly, courseExample);
+        courses = courseRepository.findAll(spec);
+
+        List<CourseDTO> courseDtos = courseMapper.toDTO(courses);
+        addRole(courseDtos);
+        courseDtos = courseDtos.stream().filter(o -> o.getCourseRole() != null).collect(Collectors.toList());
+        courseDtos = courseDtos.stream().filter(
+                o -> !(o.getCourseRole() == COURSEROLE.PARTICIPANT && o.getPublishState() == PUBLISHSTATE.UNPUBLISHED))
```

Comment:
Should not be a possible state in our application, so I would acutally suggest removing this check here.

Reply:
since the backend provides the unbublish functionality, i would leave it for the moment